Bottom: fb1504300eed5180282f8fa7fbb7430ff512f2ba
Top:    b219a4a9def6c76ad83ce9a2d4024378b8b7ad17
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-05-03 18:07:10 +0100

Refresh of hid-gtk-drop-in-gl-renderer-wo

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index ffbd788..e5c3c17 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -808,7 +808,6 @@ ghid_drawing_area_expose_cb (GtkWidget *widget,
                              GHidPort *port)
 {
   BoxType region;
-  int eleft, eright, etop, ebottom;
 
   ghid_start_drawing (port);
 
@@ -841,20 +840,10 @@ ghid_drawing_area_expose_cb (GtkWidget *widget,
   region.Y1 = MIN (Py (ev->area.y), Py (ev->area.y + ev->area.height + 1));
   region.Y2 = MAX (Py (ev->area.y), Py (ev->area.y + ev->area.height + 1));
 
-  eleft = Vx (0);  eright  = Vx (PCB->MaxWidth);
-  etop  = Vy (0);  ebottom = Vy (PCB->MaxHeight);
-
   glColor3f (port->bg_color.red / 65535.,
              port->bg_color.green / 65535.,
              port->bg_color.blue / 65535.);
 
-  glBegin (GL_QUADS);
-  glVertex3i (eleft,  etop,    0);
-  glVertex3i (eright, etop,    0);
-  glVertex3i (eright, ebottom, 0);
-  glVertex3i (eleft,  ebottom, 0);
-  glEnd ();
-
   glPushMatrix ();
   glScalef ((ghid_flip_x ? -1. : 1.) / port->zoom,
             (ghid_flip_y ? -1. : 1.) / port->zoom,
@@ -864,6 +853,13 @@ ghid_drawing_area_expose_cb (GtkWidget *widget,
                 ghid_flip_y ? port->view_y0 - PCB->MaxHeight :
                              -port->view_y0, 0);
 
+  glBegin (GL_QUADS);
+  glVertex3i (0, 0, 0);
+  glVertex3i (PCB->MaxWidth, 0, 0);
+  glVertex3i (PCB->MaxWidth, PCB->MaxHeight, 0);
+  glVertex3i (0, PCB->MaxHeight, 0);
+  glEnd ();
+
   ghid_draw_bg_image ();
 
   hidgl_init_triangle_array (&buffer);
